src/shogun/mathematics/graph/Tensor.h

```diff
@@ -68,29 +67,29 @@ namespace shogun
 				if (m_data != nullptr)
 					error("Tensor already owns data!");
 				set_shape(shape);
-				m_data = allocator_dispatch(size(), m_type);
+				m_data = malloc(size_in_bytes());
```

Comment:
should we use aligned memory?

Reply:
mmm yeah we could actually do SG_ALIGNED_MALLOC and take it from there